src/plots/xyPlot.ts

```diff
@@ -22,6 +35,95 @@ export class XYPlot<X, Y> extends Plot {
 
     this._adjustYDomainOnChangeFromXCallback = (scale) => this._adjustYDomainOnChangeFromX();
     this._adjustXDomainOnChangeFromYCallback = (scale) => this._adjustXDomainOnChangeFromY();
+
+    this._fastPanZoomOnXCallback = (scale) => this._fastPanZoomOnX(scale);
+    this._fastPanZoomOnYCallback = (scale) => this._fastPanZoomOnY(scale);
+  }
+
+  private _fastPanZoomOnX(scale: Scale<X, number>) {
+    if (!this._isAnchored) {
+      return;
+    }
+
+    var domain = scale.domain();
+    this._fastPanZoomScaleX = (scale.scale(this._fastPanZoomKnownDomainX[1]) - scale.scale(this._fastPanZoomKnownDomainX[0])) /
+                              (scale.scale(domain[1]) - scale.scale(domain[0]));
+    this._fastPanZoomDeltaX = scale.scale(this._fastPanZoomKnownDomainX[0]) - scale.scale(domain[0]);
+
+    if (this._renderArea != null) {
+      this._renderArea.attr("transform",
+        "translate(" + this._fastPanZoomDeltaX + ", " + this._fastPanZoomDeltaY + ")" +
+        "scale(" + this._fastPanZoomScaleX + ", " + this._fastPanZoomScaleY + ")");
```

Comment:
Where else are these instance variables use?  Why not wrap this function into a `_fastPanZoom` function()?

Also, this seems to be working with any type of scale updating.